### ISSUE TITLE
Missing simple request serialization

### ISSUE THREAD

**Issue by lmartellotto** [kobylynskyi/graphql-java-codegen#154]
Hello,

**Is your feature request related to a problem? Please describe.**
We use this project to test our GraphQL API with RestAssured. Our API exposes two endpoints: GET and POST, as mentionned here: https://graphql.org/learn/serving-over-http/.

We generate requests as follow:
```java
ItemByIdQueryRequest query = new ItemByIdQueryRequest.Builder()
                                                          .setId(1)
                                                          .build();
        
ItemResponseProjection fields = new ItemResponseProjection()
                                                          .name()
                                                          .tags();
GraphQLRequest request = new GraphQLRequest(query, fields);
```

... which generates GraphQL following JSON query thanks to `request.toString():`
```
{"query":"query { itemById(id: 1){ name tags } }"}
```
This JSON can be sent to the POST endpoint, it works fine. But not for the GET endpoint! Because the GET endpoint is waiting for "query" parameter where the value is not a JSON query, but the GraphQL query itself. So, we need to have only:

```
query { itemById(id: 1){ name tags } }
```

**Describe the solution you'd like**
Maybe just a new getter in the `GraphQLRequest` class, which returns the GraphQL query without JSON encapsulation. 

What about a `toString()` method overload, like:

```java
    @Override
    public String toString() {
        return toString(true);
    }

    public String toString(final boolean jsonEncapsulation) {
        // TODO
    }
```

All the serialization work is already done in the `GraphQLRequestSerializer.serialize()`; it just needs to not call the last method `buildJsonQuery()` when we don't need JSON encapsulation.

If you are ok with that, I can submit a PR :)

Thanks for your time,
Laura

**Comment by kobylynskyi** [comment]
@lmartellotto This sounds good to me!

**Comment by joffrey-bion** [comment]
That makes total sense for me as well. I would even suggest that we refrain from using the name `toString()` for the JSON transformation, but maybe that's a backward compatibility issue :/

**Comment by kobylynskyi** [comment]
@joffrey-bion actually no, it is not a backward-compatibility issue.
`toString()` method of the generated class is being called only from `GraphQLRequestSerializer`.

**Comment by kobylynskyi** [comment]
@lmartellotto I suggest having following methods in `GraphQLRequest` class:
```java

    /**
     * @deprecated Not intended for use and will be removed in the next version.
     * Please use one of: {@link #toJsonRequest} or {@link #toRequest}
     */
    @Override
    @Deprecated
    public String toString() {
        return GraphQLRequestSerializer.serialize(this);
    }

    /**
     * TODO: PROPER JAVADOC
     */
    public String toJsonRequest() {
        return GraphQLRequestSerializer.serialize(this, true);
    }

    /**
     * TODO: PROPER JAVADOC
     */
    public String toRequest() {
        return GraphQLRequestSerializer.serialize(this, false);
    }
```

**Comment by lmartellotto** [comment]
Perfect, I agree with you both! 👍

**Comment by joffrey-bion** [comment]
Thanks for hearing me out 😃 

I find that `GraphQLRequest.toRequest()` might be confusing.
What about `toHttpJsonBody()` and `toQueryString()`?

**Comment by kobylynskyi** [comment]
Agreed.
Then let it be:

```java

    /**
     * @deprecated Not intended for use and will be removed in the next version.
     * Please use one of: {@link #toHttpJsonBody} or {@link #toQueryString}
     */
    @Override
    @Deprecated
    public String toString() {
        return toHttpJsonBody();
    }

    /**
     * TODO: PROPER JAVADOC
     */
    public String toHttpJsonBody() {
        return GraphQLRequestSerializer.toHttpJsonBody(this);
    }

    /**
     * TODO: PROPER JAVADOC
     */
    public String toQueryString() {
        return GraphQLRequestSerializer.toQueryString(this);
    }
```

**Comment by kobylynskyi** [comment]
@lmartellotto are you planning to work on this feature this week? 
No rush. The reason I am asking is that I want to release [version 1.8.0](https://github.com/kobylynskyi/graphql-java-codegen/milestone/12) this week.

**Comment by kobylynskyi** [comment]
Released in [1.8.0](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v1.8.0)
